clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -374,20 +374,20 @@ public void wakeup() {
     /**
      * A collection of sensors for the sender
      */
-    private class SenderMetrics {
+    private static class SenderMetrics {
```

Comment:
Why we want to make it static? I feel it is not worthwhile as we now have to send client and metadata parameters into the class.